### PR TITLE
Add CompilerWorks UDFs

### PR DESCRIPTION
**Pull request by rajc242**
CompilerWorks UDFs are added to community UDFs.

### CODE REVIEW

File: udfs/community/cw_array_compact.sqlx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+config { hasOutput: true }
+/*
+ * Copyright 2019 Google LLC
```
Comment (danieldeleo) [inline udfs/community/cw_array_compact.sqlx:3]:
```suggestion
 * Copyright 2022 Google LLC
```

Comment (danieldeleo) [reply udfs/community/cw_array_compact.sqlx:3]: Update all other files to 2022 as well

File: udfs/community/cw_array_compact.sqlx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Implements Snowflake ARRAY_COMPACT */
+CREATE OR REPLACE FUNCTION ${self()}(a ANY TYPE) AS (
+    ARRAY(SELECT v FROM UNNEST(a) v WHERE v IS NOT NULL)
+);
```
Comment (danieldeleo) [inline udfs/community/cw_array_compact.sqlx:21]:
```suggestion
);

```

Comment (danieldeleo) [reply udfs/community/cw_array_compact.sqlx:21]: Add trailing newlines to all other files as well

**Review by shevek-google — APPROVED** [review]

File: udfs/community/test_cases.js [hunk 3]
```diff
@@ -1170,6 +1170,839 @@ generate_udf_test("jaccard", [
         expected_output: `CAST(0.25 AS FLOAT64)`
     },
 ]);
+generate_udf_test("cw_instr4", [
+    {
+        inputs: [
+            `"TestStr123456"`,
+		    `"Str"`,
+		    `CAST(1 AS INT64)`,
+            `CAST(1 AS INT64)`
```
Comment (danieldeleo) [inline udfs/community/test_cases.js:1179]:
```suggestion
            `"TestStr123456"`,
		        `"Str"`,
		        `CAST(1 AS INT64)`,
            `CAST(1 AS INT64)`
```

File: udfs/community/test_cases.js [hunk 4]
```diff
@@ -1170,6 +1170,839 @@ generate_udf_test("jaccard", [
         expected_output: `CAST(0.25 AS FLOAT64)`
     },
 ]);
+generate_udf_test("cw_instr4", [
```
Comment (danieldeleo) [inline udfs/community/test_cases.js:1173]: Please add all your tests to end of this file

**Review by danieldeleo — CHANGES_REQUESTED** [review]
Will merge after comments addressed

File: udfs/community/README.md [hunk 5]
```diff
@@ -116,6 +190,631 @@ results:
...
+### [cw_twograms(t STRING)](cw_twograms.sqlx)
+```sql
+
+```
+
+### [cw_threegrams(t STRING)](cw_threegrams.sqlx)
+```sql
+
+```
+
```
Comment (danieldeleo) [inline udfs/community/README.md:700]: these are empty

File: udfs/community/README.md [hunk 6]
```diff
@@ -116,6 +190,631 @@ results:
...
+```sql
+SELECT bqutil.fn.cw_csvld('Test#123', '#', '"', 2);
+
+["Test", "123"]
+```
+
+### [cw_json_enumerate_array(text STRING)](cw_json_enumerate_array.sqlx)
+```sql
+
+```
```
Comment (danieldeleo) [inline udfs/community/README.md:756]: empty

File: udfs/community/README.md [hunk 7]
```diff
@@ -116,6 +190,631 @@ results:
...
+
+### [cw_ts_pattern_match(evSeries ARRAY<STRING>, regexpParts ARRAY<STRING>)](cw_ts_pattern_match.sqlx)
+ts_pattern_match is function that returns range of matched pattern in given UID, SID (user session)
+```sql
+
+```
```
Comment (danieldeleo) [inline udfs/community/README.md:762]: empty

File: udfs/community/README.md [hunk 8]
```diff
@@ -116,6 +190,631 @@ results:
...
+SELECT bqutil.fn.cw_comparable_format_bigint([2, 8]);
+
+p                  2 p                  8
+```
+
+### [cw_ts_overlap_buckets()](cw_ts_overlap_buckets.sqlx)
+Merges two periods together if they overlap and returns unique id for each merged bucket. Coalesces meeting periods as well (not just overlapping periods) if includeMeets is true.
+```sql
+
+```
```
Comment (danieldeleo) [inline udfs/community/README.md:817]: empty

File: udfs/community/test_cases.js (hunk 3 above)
Comment (danieldeleo) [inline udfs/community/test_cases.js:1267]:
```suggestion
            `"TestStr123456"`,
		        `"Str"`,
		        `CAST(1 AS INT64)`,
            `CAST(1 AS INT64)`
```

**Review by danieldeleo — APPROVED** [review]

**Pull request merged**